### PR TITLE
Expose `dirname` from `SourceCodePos`

### PR DESCRIPTION
**Pull request by mneumann**
This will enable `Spec` to print absolute or project-root relative file paths.

### CODE REVIEW

**Review by jemc — APPROVED** [review]

**Comment by mneumann** [comment]
Mainly, reporting absolute paths in `Spec` will break the Spec for Spec, as it would include my absolute paths. Also, I never really liked it to expose the absolute paths of my failing tests in "bug reports". No one should see something like `/home/foobar/Clients/StupidClientA/...` if I decide to name a client like this (which I don't) and reflect this in my directory structure :)

**Comment by jemc** [comment]
I agree that having `Spec` show the relative path compared to the path of the root manifest would be the most correct behavior - I prefer this over something more magical involving the "current working directory".

**Comment by jemc** [comment]
@mneumann - I think `ctx.root_package_link.path` will get you the path to the root package's manifest directory.

**Comment by mneumann** [comment]
@jemc relative to root package might not always work in case the packages would not reside under `deps`. In case of `Spec` I would report pkgname + package relative file path.

File: core/SourceCodePos.savi [hunk 1]
```diff
@@ -2,6 +2,10 @@
   // TODO: include more properties from the original Savi::Source::Pos?
   :var string String: ""
   :var filename String: ""
+  :: The filepath relative to the package
+  :var filepath_pkgrel String: ""
   :var dirname String: ""
+  :var pkgname String: ""
+  :var pkgpath String: ""
```
Comment (jemc) [inline core/SourceCodePos.savi:9]: I think I want to eventually revisit these names before Savi 1.0 - I'd like to see a more consistent and less abbreviated naming pattern.

However, this is good enough for now, and any changes we make in the future can be smoothly deprecated and later removed before Savi 1.0

**Comment by jemc** [comment]
> relative to root package might not always work in case the packages would not reside under deps. In case of Spec I would report pkgname + package relative file path.

I'd honestly still prefer them being relative to the root package, so in the case of non-root package source files I'd be able to see which package they came from. But I'm not strongly opinionated enough to argue about it, as Savi's name requirements make collisions a lot less likely.

In the strange case you raise of dependencies not being in `deps` (which would be true of `core` for now, so I guess it's not that strange at the moment), I'd expect to see `../` segments added to the head of the path until reaching a common parent directory that holds both the root package and the out-of-source-tree package.

**Comment by mneumann** [comment]
@jemc I agree with you. Added `filepath_rootpkgrel`. We should change the naming. But this works fine so far with `Spec`. So I am going to merge and feel free to fixup later on.

**Pull request merged**